CineElorrietaMVC/src/main/java/Vista/PanelGeneros.java

```diff
@@ -2,40 +2,217 @@
+		lbl_SeleccionGenero.setBounds(61, 227, 149, 14);
+		add(lbl_SeleccionGenero);
+
+		txt_seleccion = new JTextField();
+		txt_seleccion.setBounds(61, 252, 168, 37);
+		add(txt_seleccion);
+		txt_seleccion.setColumns(10);
+
+		JScrollPane scrollPane = new JScrollPane();
+		scrollPane.setBounds(271, 63, 168, 137);
+		add(scrollPane);
+		
+		JList lista_pelis = new JList();
+		lista_pelis.setEnabled(false);
+		scrollPane.setViewportView(lista_pelis);
+		
+		DefaultListModel modelo = new DefaultListModel();//necesario crear un modelo con el que llenar el jlist
+		JButton btnAnadir = new JButton("Continuar");
+		btnAnadir.setEnabled(false);
 		
+		btnAnadir.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				controladorPanelGeneros.accionadoBotonSeleccionarPanelGeneros(Integer.parseInt(seleccion));
+				
+				
+			}
+		});
+		
+		JButton btnElegir = new JButton("Seleccionar");
+		btnElegir.setBounds(61, 304, 168, 23);
+		btnElegir.addActionListener(new ActionListener() {
+		
+
+			public void actionPerformed(ActionEvent arg0) {
+				
+				seleccion = txt_seleccion.getText();// recogemos el dato de seleccion de genero
+
+				if (seleccion.equals("1")) {
+					modelo.clear();//vaciamos el jlist
+					for (int i = 0; i < peliculas.length; i++) {
+						if (peliculas[i].getGenero() == 1) {
+							String resultado = "";
+							System.out.println(peliculas[i].getTitulo());
+							resultado += peliculas[i].getTitulo() + "\n";
+							modelo.addElement(resultado);
+							txt_seleccion.setText("Drama");
+							
+						}
+
+					}
+					btnAnadir.setEnabled(true);
+
+				}
+				if (seleccion.equals("2")) {
+					modelo.clear();
+					for (int i = 0; i < peliculas.length; i++) {
+						if (peliculas[i].getGenero() == 2) {
+							String resultado = "";
+							System.out.println(peliculas[i].getTitulo());
+							resultado += peliculas[i].getTitulo() + "\n";
+							modelo.addElement(resultado);
+							txt_seleccion.setText("Comedia");
+						
+							
+						
+						}
+
+					}
+					btnAnadir.setEnabled(true);
+				}
+				if (seleccion.equals("3")) {
+					modelo.clear();
+					for (int i = 0; i < peliculas.length; i++) {
+						if (peliculas[i].getGenero() == 3) {
+							String resultado = "";
+							System.out.println(peliculas[i].getTitulo());
+							resultado += peliculas[i].getTitulo() + "\n";
+							modelo.addElement(resultado);
+							txt_seleccion.setText("Terror");
+							
+						}
+					}
+					btnAnadir.setEnabled(true);
+
+				}
+				if (seleccion.equals("4")) {
+					modelo.clear();
+					for (int i = 0; i < peliculas.length; i++) {
+						if (peliculas[i].getGenero() == 4) {
+							String resultado = "";
+							System.out.println(peliculas[i].getTitulo());
+							resultado += peliculas[i].getTitulo() + "\n";
+							modelo.addElement(resultado);
+							txt_seleccion.setText("Ciencia ficci�n");
+						}
+					}
+					btnAnadir.setEnabled(true);
```

Comment:
Estais haciendo todo el rato lo mismo. Crear un único método que haga ese for

Reply:
Hecho el método.
He intentado hacerlo en la clase cartelera pero por el modelo de la Jlist no he podido y lo he acabado dejando en el panel.